### ISSUE TITLE
Error: <stdin>: syntax error in line 11 scanning a quoted string (missing endquote? longer than 16384?)

### ISSUE THREAD

**Issue by gauteh** [jrfonseca/gprof2dot#30]
Hi, I am getting the following error:

```
Error: <stdin>: syntax error in line 11 scanning a quoted string (missing endquote? longer than 16384?)
String starting:"boost::proto::switch_<boost::phoenix::meta_grammar, boost::proto::tag_of<boost::
Traceback (most recent call last):
  File "/usr/bin/gprof2dot", line 11, in <module>
    load_entry_point('gprof2dot==2016.10.13', 'console_scripts', 'gprof2dot')()
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 3270, in main
    dot.graph(profile, theme)
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 2998, in graph
    tooltip = function.filename,
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 3046, in node
    self.attr_list(attrs)
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 3071, in attr_list
    self.id(value)
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 3084, in id
    self.write(s)
  File "/usr/lib/python3.6/site-packages/gprof2dot.py", line 3108, in write
    self.fp.write(s)
BrokenPipeError: [Errno 32] Broken pipe
```

Profile input: [prof.zip](https://github.com/jrfonseca/gprof2dot/files/774154/prof.zip)

**Comment by jrfonseca** [comment]
The problem is that boost template functions have names with 19625 characters, but dot restricts strings to 16Kb.

This is fixed with commit cbce9969dbfe15fe4ebda4e37d0acaa1dae3b5dc, which truncates function names to a sane value, but due to these very large names, the profile is pretty illegible.  I recommend using `-s`/`--strip` option to get smaller function names.   It might even make sense to make it the default at some point, given I suspect C++ template functions will only tend to get larger and larger...

**Comment by gauteh** [comment]
Alright, thanks!
